compiler/qsc_rca/tests/udts.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+pub mod common;
+
+use common::{check_last_statement_compute_properties, CompilationContext};
+use expect_test::expect;
+
+#[test]
+fn check_rca_for_udt_constructor_with_classical_values() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+        open Microsoft.Quantum.Math;
+        Complex(0.0, 0.0)"#,
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Classical
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_udt_constructor_with_a_dynamic_value() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+        open Microsoft.Quantum.Math;
+        use q = Qubit();
+        let r = M(q) == Zero ? 0.0 | 1.0;
```

Comment:
similar note here, you could use `DrawDynamicDouble` if you wanted to skip qubits.

Reply:
For tests, I would prefer to use explicit qubit measurements as sources of dynamism to make it clear for people that might not be as familiar with the implementation details of RCA.